GitUI/FindFilePredicateProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using GitCommands;
+
+namespace GitUI
+{
+    public interface IFindFilePredicateProvider
+    {
+        Func<string, bool> Get(string name, string workingDir);
+    }
+
+    public sealed class FindFilePredicateProvider : IFindFilePredicateProvider
+    {
+        public Func<string, bool> Get(string name, string workingDir)
+        {
+            var pattern = name.ToPosixPath();
+            var dir = workingDir.ToPosixPath();
+
+            if (pattern.StartsWith(dir, StringComparison.OrdinalIgnoreCase))
+            {
+                pattern = pattern.Substring(dir.Length).TrimStart('/');
```

Comment:
possible AOORE - https://msdn.microsoft.com/en-us/library/hxthx5h6(v=vs.110).aspx#Anchor_1
Please have a test to cover

Reply:
How it can be? 
dir.Length > 0 and pattern.Lenght >= dir.Length because pattern.StartsWith(dir)

Reply:
Yes, you're absolutely correct.
I missed it, apologies.